tinygrad/tensor.py

```diff
@@ -104,6 +104,9 @@ def _masked_setitem(target:Tensor, values:Tensor, mask:Tensor, axes:Tuple[int, .
   # select from values for each True element in mask else select from self
   return mask.where(values, target)
 
+def _arange_select(target:Tensor, arange_size:sint, target_dim:sint=0):
```

Comment:
factoring out this is nice. also pick a function name that describes what it does, not how it's done. use of arange is implementation details and it would be good to separate function name from implementation (onehot_dim or similar, see if there are similar things in onnx or torch).